cli/index.ts

```diff
@@ -0,0 +1,186 @@
+  database?: string;
+  auth?: boolean;
+  _?: (string | number)[];
+  $0?: string;
+};
+
+const OPTIONS = {
+  backend: {
+    id: "b",
+    description: "Backend language",
+    message: "Which backend language would you like?",
+    choices: [
+      { name: "TypeScript (Node/Express)", value: "typescript" },
+      { name: "Python (Flask)", value: "python" },
+    ],
+  },
+  api: {
+    id: "a",
+    description: "API type",
+    message: "Which API would you like?",
+    choices: [
+      { name: "REST", value: "rest" },
+      { name: "GraphQL", value: "graphql" },
+    ],
+  },
+  database: {
+    id: "d",
+    description: "Database",
+    message: "Which database would you like?",
+    choices: [
+      { name: "PostgreSQL", value: "postgresql" },
+      { name: "MongoDB", value: "mongodb" },
+    ],
+  },
+  auth: {
+    id: "au",
+    description: "Include built-in auth features",
+    message: "Would you like built-in auth features?",
+  },
+};
+
+const parseArguments = (args: CommandLineArgs) => {
+  const { argv } = yargs(args.slice(2)).options({
+    backend: {
+      alias: OPTIONS.backend.id,
+      type: "string",
+      description: OPTIONS.backend.description,
+      choices: OPTIONS.backend.choices.map((choice) => choice.value),
+    },
+    api: {
+      alias: OPTIONS.api.id,
+      type: "string",
+      description: OPTIONS.api.description,
+      choices: OPTIONS.api.choices.map((choice) => choice.value),
+    },
+    database: {
+      alias: OPTIONS.database.id,
+      type: "string",
+      description: OPTIONS.api.description,
+      choices: OPTIONS.database.choices.map((choice) => choice.value),
+    },
+    auth: {
+      alias: OPTIONS.auth.id,
+      type: "boolean",
+      description: OPTIONS.auth.description,
+    },
+  });
+
+  return argv;
+};
+
+const promptOptions = async (options: Options) => {
+  const prompts = [];
+  if (!options.backend) {
+    prompts.push({
+      type: "list",
+      name: "backend",
+      message: OPTIONS.backend.message,
+      choices: OPTIONS.backend.choices,
+    });
+  }
+
+  if (!options.api) {
+    prompts.push({
+      type: "list",
+      name: "api",
+      message: OPTIONS.api.message,
+      choices: OPTIONS.api.choices,
+    });
+  }
+
+  if (!options.database) {
+    prompts.push({
+      type: "list",
+      name: "database",
+      message: OPTIONS.database.message,
+      choices: OPTIONS.database.choices,
+    });
+  }
+
+  if (!options.auth) {
+    prompts.push({
+      type: "confirm",
+      name: "auth",
+      message: OPTIONS.auth.message,
+      default: false,
+    });
+  }
+
+  const answers = await inquirer.prompt(prompts);
+
+  return {
+    backend: options.backend || answers.backend,
+    api: options.api || answers.api,
+    database: options.database || answers.database,
+    auth: options.auth || answers.auth,
+  };
+};
+
+const confirmPrompt = async (options: Options) => {
+  const backendChoice = OPTIONS.backend.choices.find(
```

Comment:
We still need a way to translate value -> name, so this is the best I could come up with, let me know if there is a better way

Reply:
I think this is fine, I don't expect the number of choices to grow a lot so performance isn't really a concern.

Let's use optional chaining here though 🙂 
```ts
const backendName = OPTIONS.backend.choices.find(
    (choice) => choice.value === options.backend,
)?.name;
```

Reply:
I wasn't able to get optional chaining to work for some reason
<img width="464" alt="image" src="https://user-images.githubusercontent.com/37782734/109549362-e080f780-7a9b-11eb-86b3-51a6b76c972a.png">

Reply:
Hmm that's odd. Does it work if you run it? Could be a VSCode issue, can you double check that you're using the workspace TS version?
![image](https://user-images.githubusercontent.com/30205929/109561854-b5060900-7aab-11eb-85f8-6a76702637bc.png)

Reply:
Oh, changing that worked, thanks 👍